docs/src/main/sphinx/release/release-354.md

```diff
@@ -0,0 +1,82 @@
+# Release 354 (XXXXXXXX Mar 2021)
+
+## General
+
+* Preferred write partitioning is now triggered automatically when the number
+  of written partitions exceeds or is equal to
+  ``preferred-write-partitioning-min-number-of-partitions``. The number of written
+  partitions is estimated from stats, therefore CTAS or insert source tables need to have
+  statistics collected. To forcefully enable preferred partitioning,
+  ``preferred-write-partitioning-min-number-of-partitions`` needs to be set to ``1``. 
+  To fully disable preferred write partitioning, set ``use-preferred-write-partitioning`` to ``false``.
+  ({issue}`6920`)
+* Improve performance of certain queries involving `LEFT`, `RIGHT` and `FULL JOIN` 
+  when one side of the join is known to produce a single row. ({issue}`7090`)
+* Fix failure when subqueries contain `EXCEPT` or `INTERSECT`. ({issue}`7342`)
+* Fix incorrect results when multiple similar subqueries contain `UNION` clauses that differ 
+  only in the `ALL` vs `DISTINCT` qualifier. ({issue}`7345`)
+
+## Security
+                           
+* Fix retries for OAuth 2.0 authentication in case of token expiry. ({issue}`7172`)
+
+## CLI
+
+* Support OAuth 2.0 authentication. ({issue}`7054`)
+
+## ClickHouse connector
+
+* Use correct case for name of the schema in `CREATE SCHEMA`. ({issue}`7239`)
+
+## Elasticsearch connector
+
+* Fix failure when reading single-valued fields for array types. ({issue}`7012`)
+
+## Hive connector
+
+* Fix possible S3 connection leaks when using AVRO table format. ({issue}`7178`)
+* Fix query failure when columns of a CSV table are declared as a type other than `varchar` in Glue metastore. Columns 
+  are now interpreted as `varchar` values, instead. ({issue}`7059`)
+* Rename ``hive.parallel-partitioned-bucketed-inserts`` feature config to ``hive.parallel-partitioned-bucketed-writes``. ({issue}`7259`)
+* Fix incorrect query results when reading from an incorrectly bucketed table created and registered 
+  with the metastore by Spark. ({issue}`6848`)
+  
+## Iceberg connector
+
+* Fix queries on empty tables without a snapshot id created by Spark. ({issue}`7233`)
+* Upgrade to Iceberg ``0.11.0``. It is recommended to also upgrade the version in Iceberg readers
```

Comment:
I wrote a different version of this: https://github.com/trinodb/trino/issues/7186#issuecomment-802436709